numpy/_core/src/multiarray/item_selection.c

```diff
@@ -3085,43 +3112,83 @@ PyArray_Sort(PyArrayObject *op, int axis, NPY_SORTKIND flags)
     // Zero the NPY_HEAPSORT bit, maps NPY_HEAPSORT to NPY_QUICKSORT
     flags &= ~_NPY_SORT_HEAPSORT;
 
-    sort_table = PyDataType_GetArrFuncs(PyArray_DESCR(op))->sort;
-    switch (flags) {
-        case NPY_SORT_DEFAULT:
-            sort = sort_table[NPY_QUICKSORT];
-            break;
-        case NPY_SORT_STABLE:
-            sort = sort_table[NPY_STABLESORT];
-            break;
-        default:
-            break;
-    }
-
-    // Look for appropriate generic function if no type specific version
-    if (sort == NULL) {
-        if (!PyDataType_GetArrFuncs(PyArray_DESCR(op))->compare) {
-            PyErr_SetString(PyExc_TypeError,
-                            "type does not have compare function");
+    // Look for type specific functions
+    sort_method = NPY_DT_SLOTS(NPY_DTYPE(PyArray_DESCR(op)))->sort_meth;
+    if (sort_method != NULL) {
+        PyArray_Descr *descr = PyArray_DESCR(op);
+        PyArray_DTypeMeta *dt = NPY_DTYPE(descr);
+
+        PyArray_DTypeMeta *dtypes[2] = {dt, dt};
+        PyArray_Descr *given_descrs[2] = {descr, descr};
+        PyArray_Descr *loop_descrs[2];
+        npy_intp view_offset = 0;
+        
+        if (sort_method->resolve_descriptors(
+            sort_method, dtypes, given_descrs, loop_descrs, &view_offset) < 0) {
+            PyErr_SetString(PyExc_RuntimeError,
+                            "unable to resolve descriptors for sort");
             return -1;
         }
+        
+        PyArrayMethod_SortParameters sort_params = {
+            .flags = flags,
+        };
+        context.parameters = &sort_params;
+        context.descriptors = loop_descrs;
+
+        // Arrays are always contiguous for sorting
+        npy_intp stride = PyArray_ITEMSIZE(op);
+        npy_intp strides[2] = {stride, stride};
+        NPY_ARRAYMETHOD_FLAGS method_flags = 0;
+
+        if (sort_method->get_strided_loop(
+            &context, 1, 0, strides, &strided_loop, &auxdata, &method_flags) < 0) {
```

Comment:
Same question about `method_flags` - do they need to be propagated at all?

If "in principle yes, but in practice it doesn't matter", then maybe add a comment (same for `view_offset` above).

Reply:
I'm unsure - I don't think we're using any flags right now, so I've added a comment!